File: Machine Learning A-Z Template Folder/Part 2 - Regression/Section 6 - Polynomial Regression/Polynomial_Regression/regression_template.R
# Importing data set
data_set = read.csv('Position_Salaries.csv')
data_set = data_set[2:3]

# Splitting the data in to Training set and Test Set
# installed.packages('caTools')
# library(caTools)
# set.seed(123)
# split = sample.split(data_set$Salary, SplitRatio = 2/3)
# training_set = subset(data_set, split ==TRUE)
# test_set = subset(data_set, split ==FALSE)

# "Scaling your features".
# training_set[, 2:3] =  scale(training_set[, 2:3])
# test_set[, 2:3] = scale(test_set[, 2:3])


# Fitting Regression to the Training set
# Create your Regressor

# Predicting Regression Model result
y_pred = predict(regressor, data.frame(Level = 6.5))

# Visualize Polynomial Regression Model
ggplot() + 
  geom_point(aes(x = data_set$Level, y = data_set$Salary),
             colour = 'red') +
  geom_line(aes(x = data_set$Level, y = predict(regressor, newdata = data_set)),
            colour = 'green') +
  ggtitle('Truth or Bluff (Regression Model)') +
  xlab('Level') +
  ylab('Salary')

# Visualize Polynomial Regression Model(High resolution and smooth curve)
x_grid = seq(min(data_set$Level, max(data_set$Level), 0.1))
ggplot() + 
  geom_point(aes(x = data_set$Level, y = data_set$Salary),
             colour = 'red') +
  geom_line((aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid)),
            colour = 'green') +
  ggtitle('Truth or Bluff (Regression Model)') +
  xlab('Level') +
  ylab('Salary')

File: Machine Learning A-Z Template Folder/Part 2 - Regression/Section 5 - Multiple Linear Regression/regression_template.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Fri Mar  9 16:16:59 2018

@author: aerup
"""

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values # make sure X is always matrix
y = dataset.iloc[:, 2].values # Y is always vector

"""# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)"""

# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)"""


# Fitting The Regression Model to the data set
# Create your appropriate regressor


# Predicting a new result with regression model
y_pred = regressor.predict(6.5)

# Visulalising  the regression results
X_grid = np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape(len(X_grid), 1)
plt.scatter(X, y, color = 'red')
plt.plot(X, regressor.predict(X), color='green')
plt.title('Truth or Bluff (Polynomial Regression)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

# Visulalising  the regression results (For high resolution and smoother curve)
X_grid = np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape(len(X_grid), 1)
plt.scatter(X, y, color = 'red')
plt.plot(X, regressor.predict(X), color='green')
plt.title('Truth or Bluff (Polynomial Regression)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()